Базовая стилизация всех страниц

Alex88GitHub opened the pull request:
---
:mortar_board: [Базовая стилизация всех страниц](https://up.htmlacademy.ru/htmlcss/36/user/2254559/tasks/8)

:boom: https://htmlacademy-htmlcss.github.io/2254559-sedona-36/4/

Alex88GitHub committed:
базовая стилизация

создает файл style.css и подключает файл со стилями.

Alex88GitHub committed:
Update styles.css

добавляет стили классам

Alex88GitHub committed:
Update styles.css

добавляет комментарии в CSS файл. Добавляет стиль блока преимуществ.

Alex88GitHub committed:
Базовая стилизация

Завершает написание стилей блока advantages.

Alex88GitHub committed:
Базовая стилизация

Завершает базовую стилизацию для index.html

Alex88GitHub committed:
Базовая стилизация

1. Завершает написание стилей для catalog.html
2. Добавляет и корректирует код в pagination.

Alex88GitHub committed:
Update catalog.html

добавляет большую букву "От" в соответствие с макетом

Alex88GitHub committed:
images

Переименовывает файлы изображений

4elbik commented on index.html:
```diff
@@ -50,25 +51,25 @@ <h2 class="vissually-hiden">Добро пожаловать!</h2>
-        <li class="advantages-item">
-          <h3 class="advantages-item-title">Настоящий городок</h3>
+        <li class="advantages-item-city">
+          <h3 class="advantages-item-title-city">Настоящий городок</h3>
```
Если хочешь уточнить - то лучше добавить ещё один класс к элементу. Потому что у нас вот эти блоки они на самом деле одинаковые по каким-то стилям (причем по многим). Различие есть только в двух вещах:
1) Цвет
2) Большой или маленький блок (типо где есть картинка - большой, где нет - маленький)

Итого. Для конкретно этого элемента я бы сделал класс:
`class="advantages-item advantages-item-bg-basic-3 advantages-item-big"`

`...bg-basic-3` взял из макета, там обозван цвет таким образом. Можно свое название придумать, но в конкретном этом классе будет только тема, связанная с цветом
<img width="1320" alt="image" src="https://user-images.githubusercontent.com/21055475/217997535-fd512d88-7cbc-47b1-b394-18e6a0b44ed1.png">

Alex88GitHub replied:
сделал, но для цвета и фона использовал классы "city" и "bridge". Для размера - "advantages-item-big"

4elbik replied:
> Для цвета и фона использовал "city" "bridge"

А зачем? Нам с точки зрения стилизации, точнее, когда мы читаем селекторы в CSS, должно быть предположительно понятно что конкретно мы сейчас стилизуем (не смотря в html). И вот у нас есть стилизация по селектору `.city`. Для меня это звучит как: "Ага, это класс какого-то города. Скорее всего это какая-то конкретно иконка города"

4elbik commented on styles/styles.css:
```diff
@@ -0,0 +1,331 @@
+.hero {
+  background-image: url(../images/index-background.jpg);
+  width: 1200px;
+  height: 485px;
```
Не стоит задавать фиксированную высоту. Потом можно очень неплохо огрести при доабвлении/удалении контента
Везде, где возможно хоть малейшее переполнение - ни в коем случае нельзя ставить фиксированную высоту.

Alex88GitHub replied:
Удалил height

4elbik commented on styles/styles.css:
```diff
@@ -0,0 +1,331 @@
+/* hero */
+
+.hero {
+  background-image: url(../images/index-background.jpg);
```
Нам ещё кое какого фона не хватает:
<img width="999" alt="image" src="https://user-images.githubusercontent.com/21055475/217998396-cd9a454c-9d8d-4cee-9d24-16b19754eeea.png">
Попробуй погуглить, как можно использовать сразу два фона внутри одного селектора

Alex88GitHub replied:
Добавил фон с лесенками

4elbik commented on styles/styles.css:
```diff
@@ -0,0 +1,331 @@
+ }
+ .advantages-item {
+  width: 400px;
+  height: 385px;
```
Здесь в особенности нельзя использовать строго фиксированную высоту. Давай сейчас пока для простоты заменим на `min-height`

Alex88GitHub replied:
Заменил

4elbik commented on styles/styles.css:
```diff
@@ -0,0 +1,331 @@
+  text-align: center;
+  color: #ffffff;
+  background-color: #82b3d3;
+ }
```
`.advantages-item-city` - соответственно, этот селектор и подобные, полностью удаляться

Alex88GitHub replied:
удалил

4elbik commented on styles/styles.css:
```diff
@@ -0,0 +1,331 @@
+.subscription {
+  background-image: url(../images/subscribe-background-image.jpg);
+  width: 1200px;
+  height: 414.86px;
```
Удалить `height`

Alex88GitHub replied:
Исправил

4elbik commented on styles/styles.css:
```diff
@@ -0,0 +1,331 @@
+.catalog-filter {
+  background-image: url(../images/filter-background-image.jpg);
+  width: 1200px;
+  height: 412px;
```
Удалить `height`

Alex88GitHub replied:
Исправил

4elbik reviewed (commented):
* Весь CSS нужно "причесать" согласно [стайлгайду](https://codeguide.academy/html-css.html#css)
* > Описание фоновых параметров (цвет фона, расположение, размер и повторение фонового изображения). У каждого фона должен быть контрастный цвет текста.

Нужно добавить для фонов уточняющие свойства

Alex88GitHub committed:
Update styles.css

Добавляет пробелы между блоками правил

Alex88GitHub committed:
Update styles.css

Удаляет height, добавляет min-height

Alex88GitHub committed:
Базовая стилизация

1. Немного "причесывает" оформление стилей.
2. добавляет классы city, bridge в li, h3, p

Alex88GitHub committed:
Базовая стилизация

Добавляет класс для широкого li с изображением.

Alex88GitHub committed:
Базовая стилизация

добавляет класс для фонов блоков преимуществ

Alex88GitHub committed:
Create hero-decoration.svg

добавляет файл декоративного изображения для блока hero

Alex88GitHub committed:
Update styles.css

Добавляет декоративное изображение

Alex88GitHub committed:
Update styles.css

Добавляет свойства фоновых изображений

4elbik commented on index.html:
```diff
@@ -82,7 +83,7 @@ <h2 class="advantages-service-title">Приезжайте в Седону отд
           <p class="advantages-service-item-description">Рекомендуем пожить в настоящем мотеле, всё как в кино!</p>
         </li>
-        <li class="advantages-service-item food">
+        <li class="advantages-service-item advantages-service-item-bg-food food">
```
`advantages-service-item advantages-service-item-bg-food food` -> `advantages-service-item advantages-service-item-bg-food`

Тоже самое относиться и к "Жилью" и "Сувенирам"

4elbik commented on styles/styles.css:
```diff
@@ -0,0 +1,359 @@
+  background-size: 1200px auto, 1200px auto;
+  background-repeat: no-repeat, no-repeat;
+  background-position: bottom, top;
+  background-color:#82b3d3 ;
```
<img width="753" alt="image" src="https://user-images.githubusercontent.com/21055475/218320199-58f87cbf-917c-41c3-bf52-148654b34068.png">

1) После двоеточия обязательно пробел
2) Перед точкой с запятой пробел не нужен

Alex88GitHub replied:
исправил

4elbik commented on styles/styles.css:
```diff
@@ -0,0 +1,359 @@
+}
+
+.advantages-service-item-bg-food {
+    background-color: #ffffff;
```
`.advantages-service-item-bg-food`

Слишком большой отступ для свойства (как будто здесь 4 пробела, когда во всем остальном документе - 2)

Alex88GitHub replied:
исправил

4elbik commented on styles/styles.css:
```diff
@@ -0,0 +1,359 @@
+
+/*search-results*/
+
+.search-results-text, .search-results-number {
```
Перечисляемые селекторы - каждый с новой строчки

Alex88GitHub replied:
точняк, исправил.
